### DISCUSSION TITLE
Github Desktop pull question

### DISCUSSION THREAD

**Discussion by seemedFilm** [community/community#112214] [Other Feature Feedback, Questions, & Ideas]
Hi folks,

I'm quite new and hope that this place is correct for my question.

I have a private repo in Github.
I can commit changes to it and it will be published in github.com (to check that it was published)

Now to my question:
When I've made some changes in my local files, where I'm not happy with. How is it intented to get the files from my branch from Github.com to my local disk?
Form what I've understood the Fetch should do the trick and download the changes, or missing files which I've removed on the local disk. Is that correct?

**Comment by kuciapakrystian** [comment]
yes, you're correct! To get the latest changes from your GitHub repository to your local disk, you can use the git fetch command. However, git fetch alone doesn't actually merge these changes into your local branch; it only fetches the changes from the remote repository.

To incorporate these changes into your local branch, you have a couple of options:

Merge: After fetching the changes, you can merge them into your local branch using the git merge command.

git merge origin/<branch_name>


Replace <branch_name> with the name of the branch you want to merge from.

Pull: Alternatively, you can use git pull, which is essentially a combination of git fetch and git merge. It fetches changes from the remote repository and merges them into your current branch in one step.

git pull origin <branch_name>


This command will fetch changes from the specified branch in the remote repository (origin) and merge them into your current local branch.

After performing either git merge or git pull, you may need to resolve any conflicts that arise during the merge process. Once conflicts are resolved (if any), your local branch will be updated with the changes from the remote repository.

**Reply by seemedFilm** [reply]
Hi,

Thanks for your detailed answer! Seems that also pull that I've tried doesn't download the deleted or changed content files. 

How do I geht noticed if there are any conflicts that shall be resolved?
The Log doesn't seem to have an issue:
2024-03-12T19:49:00.737Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: Reponame - Username(has token)
2024-03-12T19:49:02.051Z - info: [ui] Executing pull: git -c credential.helper= -c rebase.backend=merge pull --ff --recurse-submodules --progress origin (took 1.216s)

Still, the deleted file on my local drive, is not replaced by the same file from my Github Repo. Should this happen or is this not intented that a Pull from Github re-creates a deleted file?

**Reply by kuciapakrystian** [reply]
When you perform a pull from GitHub, deleted files should be removed from your local drive and replaced with the version from the remote repository. If this is not happening, it could indicate a conflict or an issue with your Git configuration.
To check for conflicts, you can use the "git status" command in your terminal. This will show you any files that are in conflict and need to be resolved manually. You can then use tools like "git diff" to compare the changes and resolve any conflicts.

**Reply by seemedFilm** [reply]
Git status does not show much from interest:
$ git status
On branch Webserver
Your branch is up to date with 'origin/Webserver'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        deleted:    Lagesensor.ino

Indeed, the Lagensensor.ino was deleted on my disk. (The location is on my Documentsfolder under C:\ and the whole Documents folder is covered by OneDrive)
With the git pull, I want to download the *.ino file from my Github:
$ git pull
Already up to date.

But it wont show up.

Github Desktop logs also does not show much from interest:
2024-03-13T19:38:51.896Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2024-03-13T19:38:52.252Z - info: [ui] [AppStore] loading 1 repositories from store
2024-03-13T19:38:52.252Z - info: [ui] [AppStore] found account: xxx (xxx)
2024-03-13T19:38:52.267Z - info: [ui] [BranchPruner] Last prune took place yesterday - skipping
2024-03-13T19:38:52.414Z - info: [ui] launching: 3.3.11 (Windows 10.0.19045)
2024-03-13T19:38:52.416Z - info: [ui] execPath: 'C:\Users\Patri\AppData\Local\GitHubDesktop\app-3.3.11\GitHubDesktop.exe'
2024-03-13T19:38:52.615Z - info: [ui] Subscribed 'xxx' to Alive channel
2024-03-13T19:38:59.186Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: Lagesensor - xxx (has token)
2024-03-13T19:39:00.415Z - info: [ui] Executing pull: git -c credential.helper= -c rebase.backend=merge pull --ff --recurse-submodules --progress origin (took 1.131s)

**Reply by seemedFilm** [reply]
I guess I have to switch the branches in order to receive the missing or changed files. This works
